File: GardenTracker.Web/src/types/harvest.ts
export type HarvestUnit = 'Pounds' | 'Ounces' | 'Count' | 'Bunch'

export interface Harvest {
  id: number
  bedId: number
  bedName: string
  seasonId: number
  plantVarietyId: number
  plantVarietyName: string
  plantTypeName: string
  quantity: number
  unit: HarvestUnit
  harvestDate: string
  notes?: string | null
}

export interface CreateHarvestRequest {
  bedId: number
  plantVarietyId: number
  quantity: number
  unit: HarvestUnit
  harvestDate: string
  notes?: string | null
}

export interface UpdateHarvestRequest {
  quantity: number
  unit: HarvestUnit
  harvestDate: string
  notes?: string | null
}

File: GardenTracker.Core/Entities/Harvest.cs
using GardenTracker.Core.Enums;

namespace GardenTracker.Core.Entities;

public class Harvest
{
    public int Id { get; set; }
    public int BedId { get; set; }
    public int SeasonId { get; set; }
    public int PlantVarietyId { get; set; }
    public decimal Quantity { get; set; }
    public HarvestUnit Unit { get; set; }
    public DateOnly HarvestDate { get; set; }
    public string? Notes { get; set; }

    public Bed Bed { get; set; } = null!;
    public Season Season { get; set; } = null!;
    public PlantVariety PlantVariety { get; set; } = null!;
    public ICollection<BedPlanting> SeedSavedPlantings { get; set; } = [];

    // Populated by Dapper JOINs — not persisted
    public string? BedName { get; set; }
    public string? PlantVarietyName { get; set; }
    public string? PlantTypeName { get; set; }
}
